x-pack/plugins/stack_alerts/server/rule_types/geo_containment/lib/transform_results.ts

```diff
@@ -6,64 +6,45 @@
  */
 
 import _ from 'lodash';
-import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 import type { GeoContainmentAlertInstanceState } from '../types';
 
 // Flatten agg results and get latest locations for each entity
 export function transformResults(
-  results: estypes.SearchResponse<unknown>,
+  results: any, // eslint-disable-line @typescript-eslint/no-explicit-any
   dateField: string,
   geoField: string
 ): Map<string, GeoContainmentAlertInstanceState[]> {
-  const buckets = _.get(results, 'aggregations.shapes.buckets', {});
-  const arrResults = _.flatMap(buckets, (bucket: unknown, bucketKey: string) => {
-    const subBuckets = _.get(bucket, 'entitySplit.buckets', []);
-    return _.map(subBuckets, (subBucket) => {
-      const locationFieldResult = _.get(
-        subBucket,
-        `entityHits.hits.hits[0].fields["${geoField}"][0]`,
-        ''
-      );
-      const location = locationFieldResult
-        ? _.chain(locationFieldResult)
-            .split(', ')
-            .map((coordString) => +coordString)
-            .reverse()
-            .value()
-        : [];
-      const dateInShape = _.get(
-        subBucket,
-        `entityHits.hits.hits[0].fields["${dateField}"][0]`,
-        null
-      );
-      const docId = _.get(subBucket, `entityHits.hits.hits[0]._id`);
+  const resultsMap = new Map<string, GeoContainmentAlertInstanceState[]>();
+  const boundarySplitBuckets = results?.aggregations?.shapes?.buckets ?? {};
+  for (const boundaryId in boundarySplitBuckets) {
+    if (!boundarySplitBuckets.hasOwnProperty(boundaryId)) {
+      continue;
+    }
 
-      return {
-        location,
-        shapeLocationId: bucketKey,
-        entityName: subBucket.key,
-        dateInShape,
-        docId,
-      };
-    });
+    const entitySplitBuckets = boundarySplitBuckets[boundaryId]?.entitySplit?.buckets ?? [];
+    for (let i = 0; i < entitySplitBuckets.length; i++) {
+      const entityName = entitySplitBuckets[i].key;
+      const entityResults = resultsMap.get(entityName) ?? [];
+      entityResults.push({
+        location: entitySplitBuckets[i].entityHits?.hits?.hits?.[0]?.fields?.[geoField]?.[0] ?? '',
+        shapeLocationId: boundaryId,
+        dateInShape:
+          entitySplitBuckets[i].entityHits?.hits?.hits?.[0]?.fields?.[dateField]?.[0] ?? null,
+        docId: entitySplitBuckets[i].entityHits?.hits?.hits?.[0]?._id,
+      });
+      resultsMap.set(entityName, entityResults);
+    }
+  }
+
+  // TODO remove sort
+  // legacy algorithm sorted entity hits oldest to newest for an undocumented reason
+  // preserving sort to avoid unknown breaking changes
+  resultsMap.forEach((value, key) => {
+    if (value.length > 1) {
+      // sort oldest to newest
+      resultsMap.set(key, _.orderBy(value, ['dateInShape'], ['desc', 'asc']));
```

Comment:
Can you explain why the `orders` parameter for the `_.orderBy` function changed? Previously they were `['asc', 'desc']`.

Reply:
Yes I can and this tripped me up as well

In the original implementation posted below, notice how the data set is iterated over twice. The `_.orderBy` iteration sorts by "dateInShape" ascending to descending. The `reduce` iteration creates an array per entity name and then pushes the results into that array, thus reversing the order to descending by ascending. Mind fuck
```
const orderedResults = _.orderBy(arrResults, ['entityName', 'dateInShape'], ['asc', 'desc'])
    // Get unique
    .reduce(
      (
        accu: Map<string, GeoContainmentAlertInstanceState[]>,
        el: GeoContainmentAlertInstanceState & { entityName: string }
      ) => {
        const { entityName, ...locationData } = el;
        if (entityName) {
          if (!accu.has(entityName)) {
            accu.set(entityName, []);
          }
          accu.get(entityName)!.push(locationData);
        }
        return accu;
      },
      new Map()
    );
```

Reply:
🤯 

Well, I can certainly appreciate the much cleaner implementation now. Thank you! 👍